src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java

```diff
@@ -367,6 +372,37 @@ public StudentExamResource(ExamAccessService examAccessService, StudentExamServi
         return ResponseEntity.ok(testRun);
     }
 
+    /**
+     * POST /courses/{courseId}/exams/{examId}/student-exams/automatically-assess-unsubmitted-student-exams : Automatically assess unsubmitted student exams.
+     *
+     * Finds student exams which the students did not submit on time i.e {@link StudentExam#isSubmitted()} is false.
+     * Automatically grade all modeling- and text exercises with 0 points in {@link StudentExamService#automaticallyAssessUnsubmittedExams}.
+     *
+     * @param courseId the id of the course
+     * @param examId the id of the exam
+     * @return {@link HttpStatus#BAD_REQUEST} if the exam is not over yet | {@link HttpStatus#FORBIDDEN} if the user is not an instructor
+     */
+    @PostMapping("/courses/{courseId}/exams/{examId}/student-exams/automatically-assess-unsubmitted-student-exams")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Integer> automaticallyAssessUnsubmittedStudentExams(@PathVariable Long courseId, @PathVariable Long examId) {
+        log.info("REST request to automatically assess the not submitted student exams of the exam with id {}", examId);
+
+        Optional<ResponseEntity<Void>> courseAndExamAccessFailure = examAccessService.checkCourseAndExamAccessForInstructor(courseId, examId);
+        if (courseAndExamAccessFailure.isPresent()) {
+            return forbidden();
+        }
+
+        if (!this.examService.isExamOver(examId)) {
+            // you can only grade not submitted exams if the exam is over
+            return badRequest();
+        }
```

Comment:
Looks like we fetch the exam twice here?

Reply:
@jpbernius Yeah, this is the case in most methods in the `ExamResource` & `StudentExamResource` because the public methods of the `ExamAccessService `only take the examId as a parameter and fetch the exam in the body. To my knowledge there is no way to only fetch the exam once if we want to reuse the checks of the `ExamAccessService`.

Reply:
We have a similar situation with the user object for permission checks. In this instance, we fetch the user before the checks and pass it as a reference to a second version of the check methods.

Reply:
I added methods which take the prefetched exam in the ExamAccessService and am using those now in 135dbb812d0c8afbd8cb6f60491facf58e9eae8d